src/components/WebContainerEditor.tsx

```diff
@@ -22,13 +25,20 @@ const CodeEditor = dynamic(
 export function WebContainerEditor() {
   const [output, setOutput] = useState<string[]>([])
   const [isRunning, setIsRunning] = useState(false)
+  const [aiExplanation, setAIExplanation] = useState<string>()
+  const [askingAI, setAskinAI] = useState(false)
   const { codeSnippet, setCodeSnippet } = useEditor()
+  const router = useRouter()
+
+  const id = router.query.id as string
+  const isSavedNotePage = typeof id === 'string'
 
   useEffect(() => {
     return () => {
       setOutput([])
+      setAIExplanation(undefined)
```

Comment:
What happens if we don't cleanup?

Reply:
When changing the page, the `WebContainerEditor` renders the output and the AI explanation automatically (from the last visited page), even though we don't press any button in the current route.